pipeline-runner/R/gem2s-2-load_cellranger_files.R

```diff
@@ -1,32 +1,42 @@
-#' Read input folder of 10x data
+#' Read user input files
+#'
+#' Checks technology used and dispatches call to correct reading function.
+#' In case of 10x, `annot` contains a data.frame from reading the features.tsv.gz file,
+#' while in rhapsody it contains the gene symbols as read from the count matrix.
 #'
 #' @inheritParams download_user_files
 #' @param prev_out list with experiment configuration settings
 #'
 #' @return list with 'output' slot containing \itemize{
 #'   \item{"counts_list"}{named list of dgCMatrix per sample}
-#'   \item{"annot"}{data.frame from reading features.tsv.gz for each sample}
-#' }
+#'   \item{"annot"}{data.frame with gene ids and/or symbols}
 #' @export
-#'
-load_cellranger_files <- function(input, pipeline_config, prev_out, input_dir = "/input") {
-  message("Loading cellranger output ...")
+load_user_files <- function(input, pipeline_config, prev_out, input_dir = "/input") {
+  message("Loading user files...")
   check_prev_out(prev_out, "config")
 
   # destructure previous output
   config <- prev_out$config
 
-  output <- c(prev_out, call_read10x(config, input_dir))
+  technology <- config$input$type
+
+  read_fun <- switch(technology,
+    "10x" = call_read10x,
+    "rhapsody" = call_read_rhapsody
+  )
+
+  output <- c(prev_out, read_fun(config, input_dir))
 
   res <- list(
     data = list(),
     output = output
   )
 
-  message("\nLoading of cellranger files step complete.")
+  message("\nLoading of ", technology, " files step complete.")
   return(res)
 }
 
+
 #' Calls Read10X
 #'
 #' Cellranger outputs from V2 and V3 kits were renamed to look like V3 (features.tsv.gz).
```

Comment:
on lines 51 and 52 that are not visible here, I would move the message `Samples to include in the analysis` inside `load_user_files` function (it is copied twice now in individual technology read functions)